### PR TITLE
Add LBP support to MsgCreatePool json

### PR DESCRIPTION
**Pull request by sunnya97**

### CODE REVIEW

File: x/gamm/client/cli/tx.go [hunk 1]
```diff
@@ -384,6 +385,50 @@ func NewBuildCreatePoolMsg(clientCtx client.Context, txf tx.Factory, fs *flag.Fl
...
+		targetPoolAssetCoins, err := sdk.ParseDecCoins(pool.SmoothWeightChangeParams.TargetPoolWeights)
+		if err != nil {
+			return txf, nil, err
+		}
+
+		var targetPoolAssets []types.PoolAsset
+		for i := 0; i < len(targetPoolAssetCoins); i++ {
+
+			if targetPoolAssetCoins[i].Denom != poolAssetCoins[i].Denom {
+				return txf, nil, errors.New("initial pool weights and target pool weights should have same denom order")
```
Comment (ValarDragon) [inline x/gamm/client/cli/tx.go:403]: This isn't actually enforced by the state machine, the state machine will handle sorting. Probably more convenient to also enforce here though

File: x/gamm/client/cli/tx.go [hunk 2]
```diff
@@ -384,6 +385,50 @@ func NewBuildCreatePoolMsg(clientCtx client.Context, txf tx.Factory, fs *flag.Fl
...
+
+			poolAssets = append(poolAssets, types.PoolAsset{
+				Weight: poolAssetCoins[i].Amount.RoundInt(),
+				Token:  deposit[i],
+			})
+		}
+
+		smoothWeightParams := types.SmoothWeightChangeParams{
+			Duration:           duration,
+			InitialPoolWeights: poolAssets,
```
Comment (ValarDragon) [inline x/gamm/client/cli/tx.go:414]: Should just be blank

File: x/gamm/client/cli/tx.go [hunk 3]
```diff
@@ -384,6 +385,50 @@ func NewBuildCreatePoolMsg(clientCtx client.Context, txf tx.Factory, fs *flag.Fl
...
+		}
+
+		var targetPoolAssets []types.PoolAsset
+		for i := 0; i < len(targetPoolAssetCoins); i++ {
+
+			if targetPoolAssetCoins[i].Denom != poolAssetCoins[i].Denom {
+				return txf, nil, errors.New("initial pool weights and target pool weights should have same denom order")
+			}
+
+			poolAssets = append(poolAssets, types.PoolAsset{
```
Comment (ValarDragon) [inline x/gamm/client/cli/tx.go:406]: Should be targetPoolAssets here

**Comment by ValarDragon** [comment]
Fine to merge imo once my comments are resolved

File: x/gamm/client/cli/tx.go (hunk 2 above)
Comment (sunnya97) [reply x/gamm/client/cli/tx.go:414]:
```suggestion
```

**Pull request merged**